content/articles/machine-learning-image-classification-with-javascript-and-nyckel/index.md

```diff
@@ -0,0 +1,316 @@
+                    max_size = 600,
+                    width = image.width,
+                    height = image.height;
+                if (width > height)
+                {
+                    if (width > max_size)
+                    {
+                        height *= max_size / width;
+                        width = max_size;
+                    }
+                } else
+                {
+                    if (height > max_size)
+                    {
+                        width *= max_size / height;
+                        height = max_size;
+                    }
+                }
+                canvas.width = width;
+                canvas.height = height;
+                canvas.getContext('2d').drawImage(image, 0, 0, width, height);
+                var dataUrl = canvas.toDataURL('image/jpeg');
+                var resizedImage = dataURLToBlob(dataUrl);
+
+                checkImageWithNyckel(resizedImage);
+            }
+            image.src = readerEvent.target.result;
+        }
+        reader.readAsDataURL(file);
+    }
+    else
+    {
+        alert("You must choose an image");
+        resetPage();
+    }
+}
+/* Utility function to convert a canvas to a BLOB */
+var dataURLToBlob = function (dataURL)
+{
+    var BASE64_MARKER = ';base64,';
+    if (dataURL.indexOf(BASE64_MARKER) == -1)
+    {
+        var parts = dataURL.split(',');
+        var contentType = parts[0].split(':')[1];
+        var raw = parts[1];
+
+        return new Blob([raw], { type: contentType });
+    }
+
+    var parts = dataURL.split(BASE64_MARKER);
+    var contentType = parts[0].split(':')[1];
+    var raw = window.atob(parts[1]);
+    var rawLength = raw.length;
+
+    var uInt8Array = new Uint8Array(rawLength);
+
+    for (var i = 0; i < rawLength; ++i)
+    {
+        uInt8Array[i] = raw.charCodeAt(i);
+    }
+
+    return new Blob([uInt8Array], { type: contentType });
+}
+/* End Utility function to convert a canvas to a BLOB      */
+function displayResult(response)
+{
+    resetPage();
+    $("#title").text(response.labelName);
+}
+
+function resetPage()
+{
+    $("#thinking").hide();
+    $(".upload-area").show();
+    $("h2").text("Drop JPG here");
+}
+
+
+// Sending AJAX request to Nyckel to run against ML model
+function checkImageWithNyckel(image)
+{
+    var formdata = new FormData();
+    formdata.append('file', image);
+
+    $.ajax({
+        url: 'https://www.nyckel.com/v1/functions/j3l3xdfs0fv4tec7/invoke',
+        type: 'post',
+        data: formdata,
+        contentType: false,
+        processData: false,
+        dataType: 'json',
+        success: function (response)
+        {
+            displayResult(response);
+
+            //Show the JSON response in the console
+            console.log(response);
+        },
+        error: function (response)
+        {
+            alert("Error checking image", response);
+            $("#title").show();
+            resetPage();
+        }
+    });
+}
+```

```

Comment:
I'm closing this article due to lack of flow on this part. It has important code snippets that require thorough explanations.
The author has added in-line comments but they are not enough for the reader to understand the general flow of the article!